app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -108,3 +226,97 @@ const parseUserSession = (session: cognito.CognitoUserSession): UserSession => {
     const accessToken = session.getAccessToken().getJwtToken()
     return { email, accessToken }
 }
+
+// ==============
+// === SignUp ===
+// ==============
+
+const signUp = (username: string, password: string, platform: app.Platform) =>
+    results.Result.wrapAsync(() => {
+        const params = intoSignUpParams(username, password, platform)
+        return amplify.Auth.signUp(params)
+    })
+        /** The contents of a successful response are not relevant, so we discard them. */
+        .then(result => result.map(() => null))
+        .then(result => result.mapErr(intoAmplifyErrorOrThrow))
+        .then(result => result.mapErr(intoSignUpErrorOrThrow))
+
+const intoSignUpParams = (
+    username: string,
+    password: string,
+    platform: app.Platform
+): amplify.SignUpParams => ({
+    username,
+    password,
+    attributes: {
+        email: username,
+        /** Add a custom attribute indicating whether the user is signing up from the desktop. This
+         * is used to determine the schema used in the callback links sent in the verification
+         * emails. For example, `http://` for the Cloud, and `enso://` for the desktop. */
+        /** # Safety
+         *
+         * It is necessary to disable the naming convention rule here, because the key is expected
+         * to appear exactly as-is in Cognito, so we must match it. */
```

Comment:
There re 2 doc comments next to each other. Should we merge them? Im asking cause I haven't seen this anywhere else.

Reply:
yeah i think i've seen things like this elsewhere:
```ts
/* ...
 *
 * # Safety
 *
 * ...
 */
```

Reply:
At least in the rust codebase, we keep notes on safety separate from the actual doc comments sometimes:

```rust
/// doc comment
// # Safety
// foo
```

So I was following this pattern. But I can make these into one.